feat: map container environment variables to chart

d7oc opened the pull request:
This PR is aimed to add all environment variables from docker to the helm chart.

Current TODO:

- [x] Decide how empty values should be handled.
- [x] Fix the few occurring docker environment defaults where the documentation is missing.
- [x] Decide if cron setting should be split.
- [x] Decide if memcache setting should be split
- [ ] Add secrets usage for sensitive variables

d7oc committed:
- first checkin of all environment variables

xoxys commented on charts/owncloud/templates/deployment.yaml:
```diff
@@ -48,61 +48,364 @@ spec:
+            - name: OWNCLOUD_APPS_ENABLE
+              value: {{ .Values.owncloud.apps.enable | quote }}
+            - name: OWNCLOUD_APPS_INSTALL
+              value: {{ .Values.owncloud.apps.enstall | quote }}
```
apps.**i**nstall

d7oc replied:
fixed

xoxys commented on charts/owncloud/templates/deployment.yaml:
```diff
@@ -48,61 +48,364 @@ spec:
+            - name: OWNCLOUD_MAX_UPLOAD
+              value: {{ .Values.owncloud.maxUpload | quote }}
+            - name: OWNCLOUD_MEMCACHED_ENABLED
+              value: {{ .Values.owncloud.memcachedEnabled | quote }}
```
Memcached options namespacing?  `.Values.owncloud.memcached.enabled`

d7oc replied:
Yes that's exactly why this is on the TODO list above. I'm not sure if we should split it, as some settings start with `memcached` and some with `memcache` without `d`. If we switch to name spacing we lose the direct transferability from the docker name.

xoxys commented on charts/owncloud/values.yaml:
```diff
@@ -3,7 +3,7 @@ image:
   repository: docker.io/owncloud/server
   # -- Image tag.
-  # @default -- The `appVersion` of the Chart.
+    # @default -- The `appVersion` of the Chart.
```
Wrong indention.

d7oc replied:
fixed

xoxys commented on charts/owncloud/values.yaml:
```diff
@@ -17,29 +17,543 @@ nameOverride: ""
-  volume_files: /mnt/data/files
-  volume_root: /mnt/data
+  # -- Location of the access log.
+  # @default -- /dev/stdout
```
The default is set automatically from the value. You just need to override it in some special cases with an `@default` line. Can be removed in most cases below as well.

xoxys commented on charts/owncloud/values.yaml:
```diff
@@ -17,29 +17,543 @@ nameOverride: ""
+    bucket: "owncloud"
+    # -- Class to use for the objectstore. It is recommended to keep the default (see [documentation](https://doc.owncloud.com/server/latest/admin_manual/configuration/files/external_storage/s3_compatible_object_storage_as_primary.html)).
+    # @default -- OCA\Files_Primary_S3\S3Storage
+    class: "OCA\Files_Primary_S3\S3Storage"
```
YAML needs `\\` to escape or single quotes around the entire value.

d7oc replied:
```suggestion
    class: "OCA\\Files_Primary_S3\\S3Storage"
```

d7oc commented on charts/owncloud/templates/deployment.yaml:
```diff
@@ -48,61 +48,364 @@ spec:
+            - name: OWNCLOUD_APPS_ENABLE
+              value: {{ .Values.owncloud.apps.enable | quote }}
+            - name: OWNCLOUD_APPS_INSTALL
+              value: {{ .Values.owncloud.apps.enstall | quote }}
```
```suggestion
              value: {{ .Values.owncloud.apps.install | quote }}
```

d7oc committed:
fixed typo

d7oc commented on charts/owncloud/values.yaml:
```diff
@@ -3,7 +3,7 @@ image:
   repository: docker.io/owncloud/server
   # -- Image tag.
-  # @default -- The `appVersion` of the Chart.
+    # @default -- The `appVersion` of the Chart.
```
```suggestion
  # @default -- The `appVersion` of the Chart.
```

d7oc committed:
Fixed indentation

d7oc committed:
Fixed escaping

d7oc committed:
- generated new README.md docs
- fixed outstanding DOCKER default values
- cleanup for removed values
- removed default doc comments where not needed

d7oc committed:
- updated documentation readme

xoxys commented on charts/owncloud/values.yaml:
```diff
@@ -17,29 +17,386 @@ nameOverride: ""
+    # -- List of apps to disable on container startup.
+    disable: ""
+    # -- List of apps to enable on container startup.
+    # @default -- {{ .Values.owncloud.apps.install }}
```
Would you mind adding code fences to all of those values? E.g. `` `{{ .Values.owncloud.apps.install }}` ``

d7oc replied:
Done, see https://github.com/owncloud-docker/helm-charts/pull/11/commits/cd0b5075616018ffe880a3c423c087ee189671bc and https://github.com/owncloud-docker/helm-charts/pull/11/commits/1d4239b5250756b6e2873bf90a9d770ce9baab42

xoxys reviewed (changes requested):
Just a small one, LGTM otherwise.

d7oc committed:
- added code fences to default values

d7oc committed:
- updated doc README
